src/main/java/storage/Storage.java

```diff
@@ -129,6 +131,73 @@ public void createChapter(String chapterName, String moduleName) {
         return chapters;
     }
 
+    private String retrieveChapterDeadline(String moduleName, String chapterName) {
+        File f = new File(filePath + "/" + moduleName + "/" + "dues" + "/" + chapterName + "due" + ".txt");
+        try {
+            Scanner s = new Scanner(f);
+            if (s.hasNext()) {
+                return s.nextLine();
+            } else {
+                return "Invalid Date";
+            }
+        } catch (FileNotFoundException e) {
+            return "Does not exist";
+        }
+    }
+
+    private boolean missingDueFileCheck(String moduleName, String chapterName) {
+        File f = new File(filePath + "/" + moduleName + "/" + chapterName + ".txt");
+        return f.exists();
+    }
+
+    public ArrayList<DueChapter> loadAllDueChapters() throws FileNotFoundException {
+        //Loading in Modules
+        File f = new File(filePath);
+        boolean moduleExists = f.exists();
+        if (!moduleExists) {
+            throw new FileNotFoundException();
+        }
+
+        String[] modules = f.list();
+        ArrayList<DueChapter> dueChapters = new ArrayList<>();
+        for (String module : modules) {
+
+            //Loading in Chapters for each module
+            File f2 = new File(filePath + "/" + module);
+            boolean chapterExists = f2.exists();
+            if (!chapterExists) {
+                throw new FileNotFoundException();
+            }
+            String[] chapters = f2.list();
+            if (chapters.length == 0) {
+                return dueChapters;
+            }
+            for (String chapter : chapters) {
+                String target = chapter.replace(".txt", "");
+                String deadline = retrieveChapterDeadline(module, target);
+                assert !deadline.equals(null) : "Invalid deadline retrieved";
+                if (deadline.equals("Invalid Date")) {
+                    System.out.println("\nThe chapter:");
+                    System.out.println("Module: " + module + "; " + "Chapter: " + target);
+                    System.out.println("has a corrupted deadline. Please revise it ASAP! It will be considered due!\n");
+                    dueChapters.add(new DueChapter(module, new Chapter(target, Scheduler.parseDate(deadline))));
+                } else if (deadline.equals("Does not exist")) {
+                    if (missingDueFileCheck(module, target)) {
+                        deadline = "Invalid Date";
+                        System.out.println("\nThe chapter:");
+                        System.out.println("Module: " + module + "; " + "Chapter: " + target);
+                        System.out.print("has a corrupted deadline. Please revise it ASAP!");
+                        System.out.println(" It will be considered due!\n");
```

Comment:
Perhaps could use ui,showToUser or create a method specifically for printing these in Ui?